test/KBaseDataObjectToFileUtils_server_test.py

```diff
@@ -471,12 +548,87 @@ def test_KBaseDataObjectToFileUtils_GenomeSetToFASTA_02(self):
+        genomeInfo_6 = self.getGenomeInfo('GCF_000022285.1_ASM2228v1_genomic',  6)  # Wolbachia sp. wRi
+
+        genome_ref_3 = '/'.join([str(genomeInfo_3[WSID_I]),
+                                 str(genomeInfo_3[OBJID_I]),
+                                 str(genomeInfo_3[VERSION_I])])
+        genome_ref_4 = '/'.join([str(genomeInfo_4[WSID_I]),
+                                 str(genomeInfo_4[OBJID_I]),
+                                 str(genomeInfo_4[VERSION_I])])
+        genome_ref_5 = '/'.join([str(genomeInfo_5[WSID_I]),
+                                 str(genomeInfo_5[OBJID_I]),
+                                 str(genomeInfo_5[VERSION_I])])
+        genome_ref_6 = '/'.join([str(genomeInfo_6[WSID_I]),
+                                 str(genomeInfo_6[OBJID_I]),
+                                 str(genomeInfo_6[VERSION_I])])
+
+        # upload Tree
+        genome_refs_map = { '23880/3/1': genome_ref_3,
+                            '23880/4/1': genome_ref_4,
+                            '23880/5/1': genome_ref_5,
+                            '23880/6/1': genome_ref_6
+                        }
+        obj_info = self.getTreeInfo('Tiny_things.SpeciesTree', 0, genome_refs_map)
+        tree_ref = str(obj_info[WSID_I])+'/'+str(obj_info[OBJID_I])+'/'+str(obj_info[VERSION_I])
+
+        #feature_id_0 = 'A355_RS00030'   # F0F1 ATP Synthase subunit B
+        #feature_id_1 = 'WOO_RS00195'    # F0 ATP Synthase subunit B
+        #feature_id_2 = 'AOR14_RS04755'  # F0 ATP Synthase subunit B
+        #feature_id_3 = 'WRI_RS01560'    # F0 ATP Synthase subunit B
+
+
+        genome_id_feature_id_delim = '.f:'
+
+        output_dir = os.path.join(self.scratch,'fasta_out.'+str(uuid.uuid4()))
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
+        parameters = {
+                'tree_ref':            tree_ref,
+                'file':                test_name+'-test_genomeSet.fasta',
+                'dir':                 output_dir,
+                'console':             [],
+                'invalid_msgs':        [],
+                'residue_type':        'protein',
+                'feature_type':        'CDS',
+                'record_id_pattern':   '%%genome_ref%%'+genome_id_feature_id_delim+'%%feature_id%%',
+                'record_desc_pattern': '[%%genome_ref%%]',
+                'case':                'upper',
+                'linewrap':            50,
+                'id_len_limit':        5,
+                'merge_fasta_files':   'TRUE'
+                }
+        ret = self.getImpl().SpeciesTreeToFASTA(self.getContext(), parameters)[0]
+        self.assertIsNotNone(ret['fasta_file_path_list'][0])
+        self.assertIsNotNone(ret['feature_ids_by_genome_id'])
+        self.assertNotEqual(len(ret['feature_ids_by_genome_id'].keys()), 0)
+        self.assertIsNotNone(ret['short_id_to_rec_id'])
+        self.assertNotEqual(len(ret['short_id_to_rec_id']), 0)
```

Comment:
I am not sure if this is efficient. This will only check the return values are not None. I would exam the actual return value. e.g. `self.assertCountEqual(ret['feature_ids_by_genome_id'], expected_feature_ids)` or at least `self.assertIn(some_expected_feature_ids, ret['feature_ids_by_genome_id'])` and `self.assertEqual(len(ret['short_id_to_rec_id']), expected_value)`